bzt/modules/blazemeter.py

```diff
@@ -138,13 +139,30 @@ def _impl(self, *args, **kwargs):
                 time.sleep(self._user.timeout)
                 method(self, *args, **kwargs)
                 self.log.info("Succeeded with retry")
-            except (IOError, TaurusNetworkError):
+            except NETWORK_PROBLEMS:
                 self.log.error("Fatal error sending data: %s", traceback.format_exc())
                 self.log.warning("Will skip failed data and continue running")
 
     return _impl
 
 
+def get_with_retry(method):
+    @wraps(method)
+    def _impl(self, *args, **kwargs):
+        if not isinstance(self, CloudProvisioning):
+            raise TaurusInternalException("get_with_retry should only be applied to CloudProvisioning class methods")
+
+        while True:
+            try:
+                return method(self, *args, **kwargs)
+            except NETWORK_PROBLEMS:
+                self.log.debug("Error making request: %s", traceback.format_exc())
+                self.log.warning("Failed to make request, will retry in %s sec...", self.user.timeout)
+                time.sleep(self.user.timeout)
```

Comment:
Do we really need to pause between tries? I though `timeout` was more of a "time limit for a request" rather than a "cooldown period".

Reply:
Also, `get_with_retry()` should probably have a limit for a number of tries, so it doesn't run indefinitely.

Reply:
Infinite retries is current requirement from users. Pause between retries is required to not kill BZA in case of problems.

Reply:
Still, i'm not sure it's nice to freeze Taurus completely (which is what `sleep()` does) for 30 seconds if any network request failed before retrying. Do you think we can achieve delayed retry without using `sleep()`?

Reply:
We use "sleep" approach for any network retry in our code. I just made this by analogy.

Avoiding sleep() is achievable, but would require introducing new fields into class and good amount of logic around. I don't like introducing much of a code for such a small feature.

Reply:
Alright, let's merge it then. We can think of achieving sleep-less delayed retries later.